x/hub/keeper/keeper.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibctypes "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint/types"
+
+	"github.com/dymensionxyz/dymension-rdk/x/hub/types"
+)
+
+type Keeper struct {
+	cdc           codec.BinaryCodec
+	storeKey      storetypes.StoreKey
+	channelKeeper types.ChannelKeeper
+}
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	channelKeeper types.ChannelKeeper,
+) Keeper {
+	return Keeper{
+		cdc:           cdc,
+		storeKey:      storeKey,
+		channelKeeper: channelKeeper,
+	}
+}
+
+// ExtractHubFromChannel extracts the hub from the IBC port and channel.
+// Returns nil if the hub is not found.
+func (k Keeper) ExtractHubFromChannel(
+	ctx sdk.Context,
+	hubPortOnRollapp string,
+	hubChannelOnRollapp string,
+) (*types.Hub, error) {
+	// Check if the packet is destined for a hub
+	chainID, err := k.ExtractChainIDFromChannel(ctx, hubPortOnRollapp, hubChannelOnRollapp)
+	if err != nil {
+		return nil, err
+	}
+
+	hub, found := k.GetHub(ctx, chainID)
+	if !found {
+		return nil, nil
+	}
+
+	return &hub, nil
+}
+
+// ExtractChainIDFromChannel extracts the chain ID from the channel
+func (k Keeper) ExtractChainIDFromChannel(ctx sdk.Context, portID string, channelID string) (string, error) {
+	_, clientState, err := k.channelKeeper.GetChannelClientState(ctx, portID, channelID)
+	if err != nil {
+		return "", fmt.Errorf("extract clientID from channel: %w", err)
+	}
+
+	tmClientState, ok := clientState.(*ibctypes.ClientState)
+	if !ok {
+		return "", nil
+	}
+
+	return tmClientState.ChainId, nil
+}
```

Comment:
please favour our shared utils package now for stuff like this
https://github.com/dymensionxyz/sdk-utils/blob/68179fdc62bb54adb05654ee6b9d1ede4bc1b672/utils/uibc/funcs.go#L28-L49
It's already used in both the hub and rdk on main branches